### ISSUE TITLE
Game crashes with MSVC runtime error

### ISSUE THREAD

**Issue by Raizen09** [Ochii/cso2-launcher#60]
I've got this problem recently after a blackout happened. Can someone help? 

![err](https://user-images.githubusercontent.com/47719360/54178603-641d3480-44d1-11e9-8943-77eabba8dc3e.jpg)
![err1](https://user-images.githubusercontent.com/47719360/54178623-74cdaa80-44d1-11e9-8db7-85ec57939dea.jpg)
![err2](https://user-images.githubusercontent.com/47719360/54178628-77c89b00-44d1-11e9-81c9-7741c035c39c.jpg)

I've done reinstalling every Redistributable but still the game crashes when loading.
Game Launcher is basically 0.2.2
This occured after a blackout. Before, the game plays nicely
The crazy thing about this is that it doesn't display the directory on what program crashes

**Comment by dounai2333** [comment]
go use version 0.1.14. https://github.com/Ochii/cso2-launcher/issues/40

**Comment by Raizen09** [comment]
> go use version 0.1.14. #40

I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up

**Comment by dounai2333** [comment]
> > go use version 0.1.14. #40
> 
> I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up

if you still have this problem after change 0.1.14, please continue to contact me in this thread, I can give you a win7 compatible version of 0.4.1

**Comment by Raizen09** [comment]
> > > go use version 0.1.14. #40
> > 
> > 
> > I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up
> 
> if you still have this problem after change 0.1.14, please continue to contact me in this thread, I can give you a win7 compatible version of 0.4.1

I haven't tested it yet, if you could please link it here so I can download it immediately

**Comment by Raizen09** [comment]
> > > go use version 0.1.14. #40
> > 
> > 
> > I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up
> 
> if you still have this problem after change 0.1.14, please continue to contact me in this thread, I can give you a win7 compatible version of 0.4.1

Okay I've tested it but I use 2 arguments. The -lang and -enablecustom, I think it only works if you disable the argument lines. But I kinda need them

![err3](https://user-images.githubusercontent.com/47719360/54207010-5f2da480-4514-11e9-9ac0-b99a5a0ad845.jpg)

**Comment by Raizen09** [comment]
somehow the -lang argument is broken due to the blackout, I found this out because I've reverted it to 0.2.2 and try one of the arguments to make sure what argument works. Only -enablecustom is left working, I don't know if what kind of conflict does it make to the microsoft visual c++ because I've reinstalled everything and still the error occurs

**Comment by Raizen09** [comment]
> > > go use version 0.1.14. #40
> > 
> > 
> > I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up
> 
> if you still have this problem after change 0.1.14, please continue to contact me in this thread, I can give you a win7 compatible version of 0.4.1

can you send me the win7 compatible version? error still occurs

**Comment by dounai2333** [comment]
> > > > go use version 0.1.14. #40
> > > 
> > > 
> > > I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up
> > 
> > 
> > if you still have this problem after change 0.1.14, please continue to contact me in this thread, I can give you a win7 compatible version of 0.4.1
> 
> can you send me the win7 compatible version? error still occurs

sorry for the late reply. since I live in China (GMT+8), I went to sleep soon after I replied to you. here is the link of mega: https://mega.nz/#F!aL4XUAyB password: Ouat7Jw6s53JMkk_-y7HwQ
new stands for version 0.4.1, old stands for version 0.1.14, and the remain one is version 0.2.2. versions 0.2.2 and 0.1.14 have been removed -enablecustom by me, so even if you don't add this on startup, the game will still read custom, there is only one difference between them, and that is the debugging information.
the new version has all the features of version 0.4.1. it also compatible with win7 and coded by @GEEKiDoS , of course it need to add -enablecustom, because I don't use it very often, so it is mostly just used by me for testing

**Comment by Raizen09** [comment]
Thanks man.

**Comment by Raizen09** [comment]
> > > > > go use version 0.1.14. #40
> > > > 
> > > > 
> > > > I don't know but i think in my case, this is different because this error occurs when launching the game instead of creating/joining a room. It happens when you open it and when the screen is black, it stays there for at least 15 to 20 seconds and then the error shows up
> > > 
> > > 
> > > if you still have this problem after change 0.1.14, please continue to contact me in this thread, I can give you a win7 compatible version of 0.4.1
> > 
> > 
> > can you send me the win7 compatible version? error still occurs
> 
> sorry for the late reply. since I live in China (GMT+8), I went to sleep soon after I replied to you. here is the link of mega: https://mega.nz/#F!aL4XUAyB password: Ouat7Jw6s53JMkk_-y7HwQ
> new stands for version 0.4.1, old stands for version 0.1.14, and the remain one is version 0.2.2. versions 0.2.2 and 0.1.14 have been removed -enablecustom by me, so even if you don't add this on startup, the game will still read custom, there is only one difference between them, and that is the debugging information.
> the new version has all the features of version 0.4.1. it also compatible with win7 and coded by @GEEKiDoS , of course it need to add -enablecustom, because I don't use it very often, so it is mostly just used by me for testing

Thanks man. Works nicely

**Comment by Ochii** [comment]
So the game crashes when you use both -lang and -enablecustom arguments? Also which architecture is your Windows 7 running?

**Comment by GEEKiDoS** [comment]
> So the game crashes when you use both -lang and -enablecustom arguments? Also which architecture is your Windows 7 running?

The launcher got a lot issues on old platform after you using cmake to generate the projects.

I moved all new code to the old 698a3f4 tree, most of issue is gone(I actually got it running on WinXp with [this](https://github.com/Chuyu-Team/VC-LTL) project)

**Comment by dounai2333** [comment]
> So the game crashes when you use both -lang and -enablecustom arguments? Also which architecture is your Windows 7 running?

from the 0.3.0 version, win7 can not start the game (can't load ui/cso2_ui.gfx), almost 100% of win7 will encounter this problem, but win10 does not have this problem. GEEKiDoS made several compatible versions and published them on Chinese websites, different people have different compatibility problems, but using the compatible version will not cause problems. for example, I'm using the GEEKiDoS code to read the custom folder for 0.2.2 version, and I resolved the runtime error after the replacement, but with your release, I will have 100% of this problem, it's weird, right?

**Comment by Ochii** [comment]
> > So the game crashes when you use both -lang and -enablecustom arguments? Also which architecture is your Windows 7 running?
> 
> The launcher got a lot of issues on old platform after you using cmake to generate the projects.
> 
> I moved all new code to the old [698a3f4](https://github.com/Ochii/cso2-launcher/commit/698a3f46dab02174517352951ac10dc9d7069240) tree, most of issue is gone(I actually got it running on WinXp with [this](https://github.com/Chuyu-Team/VC-LTL) project)

VC-LTL looks interesting, I'll have a better look at it later.

So CMake builds are broken in Windows 7? Are we missing some flags then? Do you know if the  CMake builds work with VC-LTL?

**Comment by GEEKiDoS** [comment]
> > > So the game crashes when you use both -lang and -enablecustom arguments? Also which architecture is your Windows 7 running?
> > 
> > 
> > The launcher got a lot of issues on old platform after you using cmake to generate the projects.
> > I moved all new code to the old [698a3f4](https://github.com/Ochii/cso2-launcher/commit/698a3f46dab02174517352951ac10dc9d7069240) tree, most of issue is gone(I actually got it running on WinXp with [this](https://github.com/Chuyu-Team/VC-LTL) project)
> 
> VC-LTL looks interesting, I'll have a better look at it later.
> 
> So CMake builds are broken in Windows 7? Are we missing some flags then? Do you know if the CMake builds work with VC-LTL?

CMake builds are weird, sometimes they works on Windows 7, but in most of times they are not.(I don't have Windows 7 test enviroment so I did't looked into it)

And we should generate a minidump while crashed, That would be nice for finding out issue on different machine/envrioment.

About VC-LTL with CMake, I don't know, never tried that, but look like it works for cmake ([Readme in VC-LTL](https://github.com/Chuyu-Team/VC-LTL#33-using-vc-ltl-in-cmake))

**Comment by Raizen09** [comment]
I solved the problem, I was translating the cso2_english.txt at the time but before the blackout happend I was playing the game. This occured after the blackout when I got something wrong in the lines in the txt file. When I used -enablecustom without the -lang, it worked and then I tested the -lang argument and it crashed. So I assumed that it has something to do with the txt file I modified, I can't recall which line has the mistake but when I replaced the modified lang txt file with my backup, everything works well again. Somehow this has been a false alarm to anyone at this point. Sorry for making such a fuss over this.
